### PR TITLE
Added Financial Integration API documentation draft

### PR DESCRIPTION
**Pull request by pparedes1**

### CODE REVIEW

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 1]
```diff
@@ -0,0 +1,1124 @@
...
+### Version
+
+1.0
+
+### Authorization
+
+The Financial Integration API works only with the new [Authentication API](https://developer.concur.com/api-reference/authentication/apidoc.html).
+
+```
+HEADER:   "Authorization: Bearer {YOUR ACCESS TOKEN}"
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Remove the term 'HEADER' and extraneous whitespace; stick with the simple output:

Comment (charlieconcur) [reply src/api-reference/fi-service/financial-integration-service.markdown]:
```
Authorization: Bearer {token}"
```

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 2]
```diff
@@ -0,0 +1,1124 @@
...
+```
+
+## GET a list of Financial Transactions based on a search criteria
+
+Making a GET request to the financial transactions API will return a list of available financial documents, ready to be processed.
+
+```
+GET   /financialintegration/fi/v1/companies/{companyId}/transactiontypes/{docType}/transactions
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Extra whitespace between GET and the URI template is unnecessary.

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 3]
```diff
@@ -0,0 +1,1124 @@
...
+```
+
+### Parameters
+
+| Name | Type | Format | Description |
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Too many bars and unnecessary whitespace. Use the Markdown table model found in the documentation template located at https://github.com/concur/developer.concur.com/blob/preview/src/document-template.markdown.

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 4]
```diff
@@ -0,0 +1,1124 @@
...
+| --- | --- | --- | --- |
+| `companyId` | `string` | `path` | **Required** Company Outtask Id in Concur |
+| `docType` | `string` | `path` | **Required** The financial document type to return. VALUES: expense; invoice; cashadvance. Only one type of transactions can be retrieved at a time |
+| `OuttaskCompanyId` | `string` | `header` | **Required** Outtask Company Id in Concur |
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Path parameters and Headers should be in separate tables -- the documentation template demonstrates.

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 5]
```diff
@@ -0,0 +1,1124 @@
...
+| `entityid` | `string` | `header` | Concur unique code for the entity |
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Technically speaking, we aren't supposed to make the entityId available -- so I'd run this through a technical review.

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 6]
```diff
@@ -0,0 +1,1124 @@
...
+| `companyId` | `string` | - | Unique identifier for the company in Concur |
+| `entityId` | `string` | - | Unique Identifier for the entity in Concur |
+| `companyUuid` | `string` | `UUID` | Unique UUID for the company in Concur |
+| `erpSystemId` | `string` | - | The external System ID that processed the document |
+| `document` | `Array` | - | The JSON financial document. Review the [FI sample documents](#examples) below |
+| `docStatus` | `string` | - | The financial document status. VALUES: READY, ACKNOWLEDGED, POSTING_CONFIRMED_SUCCESS, POSTING_CONFIRMED_FAILURE. |
+
+_Example Request:_
+
+cURL:
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: No references to specific tooling allowed. Keep it super simple.

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 7]
```diff
@@ -0,0 +1,1124 @@
...
+
+```
+curl -H "OuttaskCompanyId: {YOUR COMPANY ID}" -H "Authorization: Bearer {YOUR ACCESS TOKEN}" -X GET
+        https://us.api.concursolutions.com/financialintegration/fi/v1/companies/{CompanyID}/transactiontypes/expense/transactions?limit=15
+```
+
+_Example Response:_
+
+```
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Remove the extra whitespace and preface the code block with the language. The documentation template has the examples you can follow.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 4 above)
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: Technically speaking, should not be making the `OuttaskCompanyId` available either.

File: src/api-reference/fi-service/financial-integration-service.markdown [hunk 8]
```diff
@@ -0,0 +1,1124 @@
...
+| 111 | This document has not been acknowledged | Posting |
+| 112 | This document has been recalled | Posting |
+| 113 | Confirmation has been posted for this document | Posting |
+| 114 | Document is not in a known state | Posting |
+| 115 | This document is not of type (expense, invoice, cashadvance, payroll) | Posting |
+| 116 | This document does not exist in FI | Posting |
+| 198 | Invalid request - this same request will not work if tried again. | Posting |
+| 199 | Unknown error - please try again later | Any |
+
+### Failure Codes
```
Comment (charlieconcur) [inline src/api-reference/fi-service/financial-integration-service.markdown]: We should make a clear distinction between HTTP Status Codes and API codes. Documentation template is your friend.

**Review by charlieconcur — CHANGES_REQUESTED** [review]
Good first job -- let's work together to clean up and conform to the Documentation Template.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 1 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a8). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 2 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 3 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 4 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 5 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 6 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 7 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

File: src/api-reference/fi-service/financial-integration-service.markdown (hunk 8 above)
Comment (pparedes1) [reply src/api-reference/fi-service/financial-integration-service.markdown]: Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

**Review by pparedes1 — COMMENTED** [review]
Take a look at the revised version Imene submitted today addressing your input (last commit 991304a). For the entityID and CompanyOutask ID, Teresa’s team are working on a v2 of this API to hide those details to the customers.

**Comment by charlieconcur** [comment]
Getting real close -- excellent response time to the prior feedback.

* Filename should be: `v1.financial-integration-service.markdown`.
* We need the Jekyll front matter to be at the top of the page. Replace `# Financial Integration Service` with the following:

```
---
title: Financial Integration Service v1
layout: reference
---
```
* Related: Remove `## <a name="fis"></a>Financial Integration API`.
* Make sure to follow the flow found in the documentation template. For example:
  * All schema should be at the bottom of the page, within a `## <a name="schema"></a>Schema` section; We should have links to schema in the TOC.
  * Links in TOC for everything with `##` header + `###` for anything within the Schema section.
  * ...etc.
* There are probably more HTTP headers you need to document -- fine to add them later if you do it soon -- otherwise should pause now and add them.

* Follow all of the various heading formatting found in the documentation template rather than crafting new Markdown ideas for these, for example:

```
Nay...
_Example Request:_
_Example Response:_

Yay!
### Example
#### Request
#### Response
```
* One character for bulleted lists: `*`.
* Remove whitespace from pipe characters in tables.
* Move response codes to the top since they apply to all endpoints.
* Not readily apparent how the service codes are relayed to client code.
* Make code indents two spaces and consistent throughout.
* Spell check. ;-)
* I'm confused on the example payloads and why we don't have schema for them.

I've sent @pparedes1 an edited version of the Markdown which illustrates all of the above.

**Comment by pparedes1** [comment]
IB: First of all, thank you Charlie for bringing the page up to the Concur documentation standards.

-	CO: There are probably more HTTP headers you need to document -- fine to add them later if you do it soon -- otherwise should pause now and add them:
IB: I added the known headers in the response, no specific schema is available

-	CO: Not readily apparent how the service codes are relayed to client code.
IB: It would be used by clients to identify the reason for failure, include a retry process for some cases for example or to automatically create internal support cases with the described reasons.

-	CO: I'm confused on the example payloads and why we don't have schema for them.
IB: We don’t have the details.
-	IB: waiting for the below elements schema from the development team:
o	Payment Relevant Line Items
o	concurTransactionLineItemId

File: src/api-reference/financial-integration/v1.financial-integration.markdown [hunk 9]
```diff
@@ -0,0 +1,1228 @@
...
+
+```shell
+GET https://us.api.concursolutions.com/financialintegration/fi/v1/companies/{CompanyID}/transactiontypes/expense/transactions?limit=15
+OuttaskCompanyId: {ID}
+Authorization: Bearer {token}
+```
+#### Response
+
+```shell
+HTTP/1.1 200 OK
```
Comment (charlieconcur) [inline src/api-reference/financial-integration/v1.financial-integration.markdown]: Drop `HTTP/1.1` from the example.

File: src/api-reference/financial-integration/v1.financial-integration.markdown [hunk 10]
```diff
@@ -0,0 +1,1228 @@
...
+Content-Type: application/json
+Date: date-requested
```
Comment (charlieconcur) [inline src/api-reference/financial-integration/v1.financial-integration.markdown]: This should be `{date-requested}` since you are showing a placeholder (variable) rather than an actual true value for the header.

File: src/api-reference/financial-integration/v1.financial-integration.markdown [hunk 11]
```diff
@@ -0,0 +1,1228 @@
...
+{
+  "ids": [ "5ab9224e02e840148e7cd7d9e8e72968", "2ac9224e02e840148e7cd7d9e8e12345" ]
+}
+```
+
+#### Response
+
+> This response shows both success and failure service code examples.
+
+```shell
```
Comment (charlieconcur) [inline src/api-reference/financial-integration/v1.financial-integration.markdown:228]: See comments above.

File: src/api-reference/financial-integration/v1.financial-integration.markdown [hunk 12]
```diff
@@ -0,0 +1,1228 @@
...
+    }
+  ]
+}
+```
+
+#### Response
+
+> This response shows both success and failure service code examples.
+
+```shell
```
Comment (charlieconcur) [inline src/api-reference/financial-integration/v1.financial-integration.markdown:358]: See comments above.

**Review by charlieconcur — CHANGES_REQUESTED** [review]

**Comment by charlieconcur** [comment]
Super-super close...!

Just a few additional changes and then we will be good to go.

**Comment by pparedes1** [comment]
fixed as part of a121cacb, thanks for the guidance.

**Review by charlieconcur — APPROVED** [review]
